services/wallet/activity/activity_v2.go

```diff
@@ -45,30 +38,26 @@ type FilterDependencies struct {
 	currentTimestamp func() int64
 }
 
-// getActivityEntriesV2 queries the route_* and tracked_transactions based on filter parameters and arguments
+// getSentActivitiesEntries queries the route_* and tracked_transactions based on filter parameters and arguments
 // it returns metadata for all entries ordered by timestamp column
-func getActivityEntriesV2(ctx context.Context, deps FilterDependencies, addresses []eth.Address, allAddresses bool, chainIDs []wCommon.ChainID, filter Filter, offset int, limit int) ([]Entry, error) {
+func getSentActivitiesEntries(ctx context.Context, deps FilterDependencies, addresses []eth.Address, chainIDs []wCommon.ChainID, offset int, limit int) ([]Entry, error) {
```

Comment:
hmm the reason to do a more complex SQL query, but a single one, was to be able to do filtering + paging. 
This probably works because you're taking the fetched entries as your "base" and only using the sent entries to replace, but if you perform a union (as I think we should), making 2 independent sql queries and combining them in go will cause a misbehavior.

Reply:
well that would be solved by adding a UNIQUE tx_hash to the query or something, if for some Tx type you need to get additional data for a given Tx to be able to construct the Entry you can perform additional queries.
That would be the option where we get everything on the fly, if we wanted to make the query simpler we could have a 2nd table with "post processed" entries. However you need to implement some mechanism to clear the table and re-process the Alchemy transfers if we change the processing algorithm.

Reply:
@dlipicar, I've fixed the issue with recently sent transactions before read this comment and used another approach - additional query to sent transactions table. Probably it is not the best way and one-query way would be more performant. If you would prefer this way, what would you think if I implement this as a follow-up PR? This one already takes too much time and becomes too big.